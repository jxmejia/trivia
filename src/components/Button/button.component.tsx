import { memo } from "react";
import type { ButtonProps } from "./button.definition";
import { StyledButton } from "./button.style";

const Button = ({ children, onClick }: ButtonProps) => {
  return (
    <StyledButton type="button" onClick={onClick}>
      {children}
    </StyledButton>
  );
};

export default memo(Button);
