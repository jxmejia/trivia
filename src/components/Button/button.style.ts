import styled from "@emotion/styled";

export const StyledButton = styled.button`
  font-size: 14px;
  text-transform: uppercase;
`;
