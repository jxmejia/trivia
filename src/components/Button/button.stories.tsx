import { action } from "@storybook/addon-actions";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import Button from "./button.component";

export default {
  title: "Button",
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: "Trivia challenge",
  onClick: action("Button onClick has been triggered"),
};
