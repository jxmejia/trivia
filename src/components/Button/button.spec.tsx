import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import Button from "./button.component";

expect.addSnapshotSerializer(createSerializer());

it("renders component", () => {
  const { container } = render(<Button />);
  expect(container).toMatchSnapshot();
});
