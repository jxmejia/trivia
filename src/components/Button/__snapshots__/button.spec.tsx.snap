// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is renders component 1`] = `
.emotion-0 {
  font-size: 14px;
}

<div>
  <button
    class="emotion-0"
    type="button"
  />
</div>
`;

exports[`Button renders component 1`] = `
.emotion-0 {
  font-size: 14px;
}

<div>
  <button
    class="emotion-0"
    type="button"
  />
</div>
`;

exports[`renders component 1`] = `
.emotion-0 {
  font-size: 14px;
}

<div>
  <button
    class="emotion-0"
    type="button"
  />
</div>
`;
